tasmota/tasmota_xsns_sensor/xsns_114_amsx915.ino

```diff
@@ -0,0 +1,251 @@
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef USE_I2C
+#ifdef USE_AMSX915
+/*********************************************************************************************\
+ * AMS5915, AMS6915 - Pressure and Temperature
+ *
+ * Source: Bastian Urschel
+ *
+ * I2C Address: 0x28
+\*********************************************************************************************/
+
+#define XSNS_114            114
+#define XI2C_86              86  // See I2CDEVICES.md
+
+#ifndef AMSX915_ADDR
+  #define AMSX915_ADDR         0x28
+#endif
+
+#define PMIN_DEFAULT      0
+#define PMAX_DEFAULT      0
+
+#ifndef USE_UFILESYS
+#warning "AMSx915 pressure settings cannot be saved persistent due to missing filesystem"
+#endif
+
+/********************************************************************************************/
+
+struct amsx915_data {
+  float pressure;
+  float temperature;
+  bool present;
+} amsx915data;
+
+const uint16_t AMSX915_VERSION = 0x0100;       // Latest sensor version (See settings deltas below)
+
+// Global structure containing driver saved variables
+struct {
+  uint32_t  crc32;    // To detect file changes
+  uint16_t  version;  // To detect driver function changes
+  int16_t   pmin;
+  int16_t   pmax;
+} amsx915Settings;
+
+void AmsDetect(void) {
+  // sensor datasheets does not provide any information/commands on sensor
+  // specific register so it is not possible to properly detect sensor type
+  if(Wire.available() != 4) {
+    // Error
+    amsx915data.present = false;
+    AddLog(LOG_LEVEL_INFO, PSTR("CFG: AMS not detected!"));
+
+    return;
+  }
+  while(Wire.available()) {
+    Wire.read();
+  }
+  amsx915data.present = true;
+  I2cSetActiveFound(AMSX915_ADDR, "AMSx915");
+    AddLog(LOG_LEVEL_INFO, PSTR("CFG: AMS detected"));
+}
+
+bool AmsCommand() {
+  int32_t vals[2];
+  ParseParameters(2,(uint32_t *)vals);
+  if(XdrvMailbox.data_len >= 3 && XdrvMailbox.data_len < 13) {
+    if (vals[0] >= -32768 && vals[0] < 32768) {
+      if (vals[1] >= -32768 && vals[1] < 32768) {
+        amsx915Settings.pmin = (int16_t)vals[0]; // save pmin of sensor
+        amsx915Settings.pmax = (int16_t)vals[1];  // same with pmax
+        AmsSettingsSave();
+        Response_P(S_JSON_SENSOR_INDEX_SVALUE, XSNS_114, "pressure range set");
+        return true;
+      }
+    }
+    Response_P(S_JSON_SENSOR_INDEX_SVALUE, XSNS_114, "invalid pressure range [-32768..32767]");
+    return false;
+  }
+  return false;
+}
+
+void AmsReadData(void) {
+  uint8_t buffer[4];
```

Comment:
Why don't you test for `amsx915data.present`? It is not good to send a request on the I2C bus if there is no device connected. Did I miss something?

Reply:
Think about the following case: it is possible that sensor is detached from bus (by user/contact problem/emi). If I check for `amsx915data.present` and sensor is temporary fallen of the bus, there will be no redetection of sensor. Because i2c is a master-slave-bus, it is necessary to poll the specific device address as sensors are slaves. If we won't do this, we can not redetect sensor and we need to reset the complete software which is not good at standalone measuring devices. For example the BME sensors are redetected after I remove and reattach them. I wanted to have this exact behavior at AMSx915 implementation.
I will add a counter like @barbudor showed in ens210 implementation to send requests only n seconds.

Reply:
I understand your point but 
1) Tasmota is not meant to detect sensors that are added dynamically
2) One good reason for that is you can destroy sensor or ESP by doing so

So if you want to allow redetecting a sensor that was there at the beginning but temporarily disappeared, you have to 
- Detect it once at start and have an indicator of detection (I suggest using the pointer for the dynamically allocated memory as such, no need for a `xxxx.present` boolean)
- As recommend, use that indicator to ignore totally the driver is the sensor was not detected at boot

That will prevent to poll the I2C bus for nothing it the sensor was not there at the beginning and still allow to redetected if it temporarilly disapear.

People tends to build generic binaries that include all kind of sensors they may use in order to have 1 binary they can use on different boards.
Acting as you do means that this driver is using RAM and CPU even when on a board where this sensor is not present.

Just to make it clear, I think you understand from @s-hadinger comments and mine this is a hard requirement to have the sensor merged in Tasmota